### ISSUE TITLE
Adding TinyMCE Plugins

### ISSUE THREAD

**Issue by irdublu** [cofoundry-cms/cofoundry#506]
Is it possible to add TinyMCE Plugins to enhance the existing offerings?  I'm needing Forecolor, Backcolor and Preview.  I've looked at the Installing Plugins and HTML docs pages and have done some experimenting without success.  I do not understand if TinyMCE Plugins can be included and, if they can, I don't know how.

**Comment by HeyJoel** [comment]
Have you looked at [customizing the TinyMCE config](https://www.cofoundry.org/docs/content-management/data-model-annotations/html#customizing-the-tinymce-config)? Any custom config fields you supply overwrite the default config, so you should be able to add your custom plugin selection and toolbar configuration there. E.g. I could add this via a json file like so:

**ExampleDataModel .cs**

```csharp
public class ExampleDataModel : ICustomEntityDataModel
{
    [Html(ConfigFilePath = "/content/html-editor-config.json")]
    public string Content { get; set; }
}
```

**html-editor-config.json**

```json
{
  "toolbar": "preview forecolor backcolor",
  "plugins": "preview textcolor",
}
```

Doing so will overwrite the existing plugin configuration, so you'll want to include any other plugin references you may need. The default Cofoundry selection is `link image media fullscreen imagetools code`.

**Comment by irdublu** [comment]
Yes, I found and read the HTML page and tried many things that were wrong, hence needing to raise my question, but I had no idea how simple you have made it for us.  I've used a HtmlEditorConfigSource class dictionary and it works.  Thank you very much.

**Comment by HeyJoel** [comment]
Perhaps the documentation is not clear/detailed enough, would an example similar to above have helped?

**Comment by irdublu** [comment]
Definitely, yes.

**Comment by HeyJoel** [comment]
Docs updated, will be published with the release of Cofoundry v0.11
